src/controllers/teams.controller/addTeamDocument/index.js

```diff
@@ -3,56 +3,72 @@ const catchAsync = require('../../../utils/catchAsync');
 const { query: Hasura } = require('../../../utils/hasura');
 const { add_TeamDocument } = require('./queries/mutations');
 const { checkIfAdmin } = require('./queries/queries');
+const upload = require('../../../utils/uploadDocument');
 const notify = require('../../../utils/notify');
 
 const addTeamDocument = catchAsync(async (req, res) => {
-  const sanitizerErrors = validationResult(req);
-  if (!sanitizerErrors.isEmpty()) {
-    return res.status(400).json({
-      success: false,
-      ...sanitizerErrors,
+  try {
+    const sanitizerErrors = validationResult(req);
+    if (!sanitizerErrors.isEmpty()) {
+      return res.status(400).json({
+        success: false,
+        errors: sanitizerErrors.array(),
+      });
+    }
+
+    const { cognito_sub, team_id, name } = req.body;
+
+    // Upload the document
+    const uploadMiddleware = upload('team-documents', team_id);
+    await uploadMiddleware.single('file')(req, res);
+
+    // Check if a file was uploaded
+    if (!req.file) {
+      return res.status(400).json({
+        success: false,
+        errorMessage: 'No file uploaded',
+      });
+    }
+
+    const url = req.file.location;
+    const mime_type = req.file.mimetype;
+
+    // Check if current user is team admin
+    const response1 = await Hasura(checkIfAdmin, {
```

Comment:
update the checkIfAdmin query to getUserId query since we are only using user id from response1.